pkg/segment/query/processor/dataprocessor.go

```diff
@@ -109,6 +108,15 @@ func (dp *DataProcessor) Fetch() (*iqr.IQR, error) {
 	}
 }
 
+func (dp *DataProcessor) IsGenerateDataProcessor() bool {
```

Comment:
nit: rename to something like `IsGenerator` or `IsDataGenerator` (sorry I missed this on my first pass)

Reply:
addressed in the latest commit.